core-common/src/main/java/org/zalando/nakadi/partitioning/PartitionStrategy.java

```diff
@@ -3,16 +3,17 @@
 import org.json.JSONObject;
 import org.zalando.nakadi.domain.EventType;
 import org.zalando.nakadi.exceptions.runtime.PartitioningException;
-
 import java.util.List;
 
-@FunctionalInterface
 public interface PartitionStrategy {
 
     String HASH_STRATEGY = "hash";
     String USER_DEFINED_STRATEGY = "user_defined";
     String RANDOM_STRATEGY = "random";
 
-    String calculatePartition(EventType eventType, JSONObject event, List<String> partitions)
+    String calculatePartition(PartitioningData partitioningData, List<String> partitions)
             throws PartitioningException;
+
+    PartitioningData getDataFromJson(EventType eventType, JSONObject jsonEvent)
```

Comment:
Why should `getDataFromJson` be part of `PartitionStrategy` interface? 
If it has to be, then better name it `getPartitionDataFromJson`. 
But I would rather have this as a `fromJson` method in the `PartitionData` class.

Reply:
Ok, I understand that this method is actually calculating the partition when I reached the implementation. 
I think we have to discuss this.